src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -399,15 +448,21 @@ FindTranslatedVar(List *appendRelList, Oid relationOid, Index relationRteIndex,
  * RestrictionEquivalenceForPartitionKeys uses both relation restrictions and join restrictions
  * to find as much as information that Postgres planner provides to extensions. For the
  * details of the usage, please see GenerateAttributeEquivalencesForRelationRestrictions()
- * and GenerateAttributeEquivalencesForJoinRestrictions()
+ * and GenerateAttributeEquivalencesForJoinRestrictions().
+ *
+ * The function also accepts the allAttributeEquivalenceList as a parameter. If that is
+ * provided externally (i.e., not NIL), the function uses it and skips generating it from
+ * the given planner context. This provides some flexibilty around how to call the function.
+ * It is especially useful if you have a pre-computed attribute equivalances, which saves
+ * us many CPU cycles.
  */
 bool
 RestrictionEquivalenceForPartitionKeys(PlannerRestrictionContext *
-									   plannerRestrictionContext)
+									   plannerRestrictionContext,
+									   List *allAttributeEquivalenceList)
```

Comment:
I'm a bit confused by the extra parameter that completely changes the behaviour of the function. Should this be a separate function?